src/components/alert/alert.tsx

```diff
@@ -180,11 +180,13 @@ export class Alert implements OpenCloseComponent, LocalizedComponent {
         <calcite-icon icon="x" scale={this.scale === "l" ? "m" : "s"} />
       </button>
     );
-    const formatter = createLocaleNumberFormatter(
-      this.effectiveLocale,
-      this.numberingSystem,
-      "always"
-    );
+    const formatter = new Intl.NumberFormat(this.effectiveLocale, {
```

Comment:
As a future enhancement, we may want to consider caching more formatters and using the whole options object as an identifier.